core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1500,6 +1504,37 @@ private void handleOffsetFetch(@Nullable SqlNode offset, @Nullable SqlNode fetch
     return node;
   }
 
+  private void validatePercentileFunctions(SqlCall call) {
+    if (call.getOperandList().size() == 2) {
+      SqlBasicCall sqlBasicCall = null;
+      SqlNodeList list = null;
+      SqlNode node1 = call.getOperandList().get(0);
+      SqlNode node2 = call.getOperandList().get(1);
+      if (node1 instanceof SqlBasicCall) {
+        sqlBasicCall = (SqlBasicCall) node1;
+      } else if (node2 instanceof SqlBasicCall) {
+        sqlBasicCall = (SqlBasicCall) node2;
+      }
+      if (node1 instanceof SqlNodeList) {
+        list = (SqlNodeList) node1;
+      } else if (node2 instanceof SqlNodeList) {
+        list = (SqlNodeList) node2;
+      }
+
+      if (sqlBasicCall != null && list != null
+          && sqlBasicCall.getOperator() instanceof SqlBasicAggFunction) {
+        SqlBasicAggFunction agg = (SqlBasicAggFunction) sqlBasicCall.getOperator();
+        if (agg.isPercentile()) {
+          // Validate that Percentile function have a single ORDER BY expression
+          if (list.getList().size() != 1) {
```

Comment:
@julianhyde I am validating here that the order by clause must only have a single expression. Is there a way to validate the type of that expression? I spent some time but could not find a way to get the type from SqlNode. Or should this be done elsewhere?

Reply:
I figured it out. Let me know if the new place for validating this seems ok.

Reply:
I don't feel strongly where you put the validation. Use an agile approach: Make sure you add a test that a bad query causes the right error to be thrown. Then the validation code will be exercised, and we can safely refactor it later.

Reply:
I used a different strategy to validate percentile functions. It does not involve adding a new field in the SqlBasicAggFunction